packages/core/src/client.ts

```diff
@@ -4,22 +4,30 @@ import { OrganizationController } from './controllers/organization';
 import { UserController } from './controllers/user';
 import { FileController } from './controllers/file';
 import { AppController } from './controllers/app';
-import { PaginationMetadata } from '@slashauth/types';
 
-type ResponseMeta<T> = Omit<rm.IRestResponse<T>, 'result'>;
-type ErrorMessage = any;
+type ErrorMessage = string;
 
-export type SlashauthResponse<ResponseData> = [
-  ResponseData | null,
-  ResponseMeta<ResponseData>,
-  ErrorMessage | null
-];
+type PageInfo = {
+  cursor?: string;
+  hasMore: boolean;
+};
 
-export type SlashauthResponseWithPagination<ResponseData> = [
-  ResponseData | null,
-  ResponseMeta<ResponseData> & PaginationMetadata,
-  ErrorMessage | null
-];
+type Metadata = {
+  error: ErrorMessage | null;
+  headers: Object;
+  statusCode: number;
+};
+
+export type SlashauthResponse<ResponseData> = {
+  data?: ResponseData;
+} & Metadata;
+
+export type SlashauthResponseWithPagination<ResponseData> = {
+  data?: {
+    data: ResponseData | PageInfo | null;
```

Comment:
not a fan of this stuttering. I'd like to have it return the object below to be closer to gql. I wasn't able to get it there (yet). will take another crack in the morning
```
data: {
   users: [], // change key based on paginated call
   pageInfo: {}
}
```

Reply:
I think you could just do away with the second structure... or remove the initial `data` field in general. The former sounds better but it would mean a lot more typing because you'd have to have each response look different as it would define a different type (e.g. users would look different than apps)

Reply:
Out of curiosity, why do you have `data` potentially returning `PageInfo`?

You could solve this with renaming this field to `paginatedResponse`

Reply:
updated to `paginatedResponse`